server/xpub-server/entities/user/resolvers.test.js

```diff
@@ -20,27 +20,21 @@ describe('User', () => {
 
     it('creates and returns a new user', async () => {
       const response = await Query.currentUser({}, {}, { user: profileId })
-      expect(response).toMatchObject({
-        identities: [
-          {
-            type: 'elife',
-            identifier: profileId,
-            aff: 'Tech team, University of eLife',
-            email: 'f72c502e0d657f363b5f2dc79dd8ceea',
-            meta: {
-              firstName: 'Tamlyn',
-              lastName: 'Rhodes',
-            },
-          },
-        ],
-      })
+      const idents = response.identities
+      expect(idents).toHaveLength(1)
+      expect(idents[0].type).toBe('elife')
+      expect(idents[0].identifier).toBe(profileId)
+      expect(idents[0].aff).toBe('Tech team, University of eLife')
+      expect(idents[0].email).toBe('f72c502e0d657f363b5f2dc79dd8ceea')
+      expect(idents[0].meta.firstName).toBe('Tamlyn')
+      expect(idents[0].meta.lastName).toBe('Rhodes')
```

Comment:
You shouldn't need to change this. `toMatchObject` will ensure that that the expectation is a subset of the actual, so in effect we can have extra fields returned in the response and the tests would pass as long as at least the expected fields are there.

Reply:
That was not my experience - try it for yourself. It might be the sub-object `meta` that prevented this.

Reply:
I tested it locally, and I get the same behaviour for `toMatchObject` as per your implementation. Happy for it to go in as is.

Reply:
See discussions online about `toMatchObject` not working for nested objects:
https://stackoverflow.com/questions/45692456/whats-the-difference-between-tomatchobject-and-objectcontaining

Interestingly there was an issue raised on the jest repo & I would have thought the PR associated with this issue: https://github.com/facebook/jest/issues/2506 would have solved it :thinking: